File: predict/src/main/java/com/emarsys/predict/di/PredictDependencyContainer.java
package com.emarsys.predict.di;

import com.emarsys.core.di.DependencyContainer;
import com.emarsys.core.endpoint.ServiceEndpointProvider;
import com.emarsys.core.storage.Storage;
import com.emarsys.predict.PredictInternal;

public interface PredictDependencyContainer extends DependencyContainer {

    PredictInternal getPredictInternal();

    PredictInternal getLoggingPredictInternal();

    Runnable getPredictShardTrigger();

    ServiceEndpointProvider getPredictServiceProvider();

    Storage<String> getPredictServiceStorage();
}

File: predict/src/main/java/com/emarsys/predict/di/PredictDependencyContainer.kt
package com.emarsys.predict.di

import com.emarsys.core.di.DependencyContainer
import com.emarsys.core.endpoint.ServiceEndpointProvider
import com.emarsys.core.storage.Storage
import com.emarsys.predict.PredictInternal

interface PredictDependencyContainer : DependencyContainer {
    fun getPredictInternal(): PredictInternal

    fun getLoggingPredictInternal(): PredictInternal

    fun getPredictShardTrigger(): Runnable

    fun getPredictServiceProvider(): ServiceEndpointProvider

    fun getPredictServiceStorage(): Storage<String?>
}